caddyhttp/httpserver/plugin.go

```diff
@@ -292,12 +292,13 @@ func (h *httpContext) MakeServers() ([]caddy.Server, error) {
 			if QUIC {
 				return nil, fmt.Errorf("cannot enable TLS client authentication with QUIC, because QUIC does not yet support it")
 			}
-			// this must be enabled so that a client cannot connect
+			// this must be false so that a client cannot connect
 			// using SNI for another site on this listener that
 			// does NOT require ClientAuth, and then send HTTP
 			// requests with the Host header of this site which DOES
-			// require client auth, thus bypassing it...
-			cfg.StrictHostMatching = true
+			// require client auth, thus bypassing it.
+			// if set to true via `insecure_disable_sni_matching` directive, the behaviour above is allowed
+			cfg.TLS.InsecureDisableSNIMatching = false
```

Comment:
Does using `insecure_disable_sni_matching` in your Caddyfile work with this line present? 🤔  My brain-that-needs-to-go-to-sleep is telling me that this overwrites any `true` value that may have been set.

Reply:
hey, yeah... it works. [here](https://github.com/caddyserver/caddy/pull/3075/files#diff-e74f72185700e304b2047c4af2ad1a48R449) I'm defaulting to the same *must* match behaviour. So, if `insecure_disable_sni_matching` is not set, the default will continue to be a strict host matching

Reply:
wait... your-brain-that-needs-to-go-to-sleep beats my brain-that-just-woke-up.